internal/allocator/allocator.go

```diff
@@ -61,15 +71,39 @@ func New() *Allocator {
 	}
 }
 
+// GetSvcIPFamily returns the service address family based on spec ClusterIPs or ClusterIP.
+func (a *Allocator) GetSvcIPFamily(clusterIPs []string) (IPFamily, error) {
+	ip := net.ParseIP(clusterIPs[0])
+	switch len(clusterIPs) {
+	case 1:
+		if ip.To4() != nil {
+			return IPv4, nil
+		} else {
+			return IPv6, nil
+		}
+	case 2:
+		ip2 := net.ParseIP(clusterIPs[1])
+		if ip2 == nil {
+			return Invalid, fmt.Errorf("Invalid address %q", clusterIPs[1])
```

Comment:
Mention the name of the function in the error, it will make the debugging easier.